tap_bing_ads/fetch_ad_accounts.py

```diff
@@ -0,0 +1,104 @@
+import requests
+import logging
+import xmltodict
+
+from requests.exceptions import HTTPError
+
+logger = logging.getLogger(__name__)
+
+SOAP_CUSTOMER_MANAGEMENT_URL = "https://clientcenter.api.bingads.microsoft.com/Api/CustomerManagement/v13/CustomerManagementService.svc"
+SOAP_REQUEST_CUSTOMER_INFO = """
+<s:Envelope xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
+    <s:Header xmlns="https://bingads.microsoft.com/Customer/v13">
+        <Action mustUnderstand="1">GetCustomersInfo</Action>
+        <AuthenticationToken i:nil="false">{authentication_token}</AuthenticationToken>
+        <DeveloperToken i:nil="false">{developer_token}</DeveloperToken>
+    </s:Header>
+    <s:Body>
+        <GetCustomersInfoRequest xmlns="https://bingads.microsoft.com/Customer/v13">
+            <CustomerNameFilter i:nil="false"></CustomerNameFilter>
+            <TopN>1</TopN>
+        </GetCustomersInfoRequest>
+    </s:Body>
+</s:Envelope>
+"""
+SOAP_REQUEST_CUSTOMER_AD_ACCOUNTS = """
+<s:Envelope xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
+    <s:Header xmlns="https://bingads.microsoft.com/Customer/v13">
+        <Action mustUnderstand="1">GetAccountsInfo</Action>
+        <AuthenticationToken i:nil="false">{authentication_token}</AuthenticationToken>
+        <DeveloperToken i:nil="false">{developer_token}</DeveloperToken>
+    </s:Header>
+    <s:Body>
+        <GetAccountsInfoRequest xmlns="https://bingads.microsoft.com/Customer/v13">
+            <CustomerId i:nil="false">{customer_id}</CustomerId>
+            <OnlyParentAccounts>false</OnlyParentAccounts>
+        </GetAccountsInfoRequest>
+    </s:Body>
+</s:Envelope>
+"""
+
+
+def _get_request_response(headers: dict, data: str):
+    try:
+        response = requests.post(
+            SOAP_CUSTOMER_MANAGEMENT_URL, headers=headers, data=data,
+        )
+
+        response.raise_for_status()
+
+        response = response.content.decode("utf-8")
+        response = xmltodict.parse(response)
+
+        return response
+    except HTTPError as http_err:
+        logger.error(f"HTTP error occurred: {http_err}")
+    except UnicodeError as unicode_err:
+        logger.error(f"Unicode decoding error occurred: {unicode_err}")
+    except Exception as err:
+        logger.error(f"Other error occurred: {err}")
+
+    raise Exception("Could not process request response")
```

Comment:
remove all the try-catches in favour of something like this: https://github.com/singer-io/singer-python/blob/master/singer/utils.py#L222